Source/ProgressionSystemRuntime/Private/LevelActors/PSStarActor.cpp

```diff
@@ -139,3 +142,42 @@ void APSStarActor::SetStartTimeMenuStars()
 
 	StartTimeMenuStarsInternal = World->GetTimeSeconds();
 }
+
+//  Is get called when a Star actor is initialized
+void APSStarActor::OnInitialized(const FVector& PreviousActorLocation)
+{
+	const FPSRowData& CurrentRowData = UPSWorldSubsystem::Get().GetCurrentRow();
+
+	// set offset from previous if it's not first
+	// if the PreviousActorTransform is empty this means it is a first element and set init transform
+	if (PreviousActorLocation.Equals(FVector::ZeroVector))
+	{
+		SetActorTransform(CurrentRowData.StarActorTransform);
+	}
+	else
+	{
+		SetActorTransform(CurrentRowData.StarActorTransform);
+		SetActorLocation(PreviousActorLocation + CurrentRowData.OffsetBetweenStarActors);
+	}
+}
+
+//  Updates star actors Mesh material to the Locked Star, Unlocked or partially achieved
+void APSStarActor::UpdateStarActorMeshMaterial(UMaterialInstanceDynamic* StarDynamicProgressMaterial, float AmountOfStars, bool bIsLockedStar)
+{
+	if (!bIsLockedStar) // unlocked stars
```

Comment:
- To improve readability and reduce nesting, early return statements can be introduced to handle conditions instead of using multiple if-else branches.
- Also, we should verify that pointers like `StarDynamicProgressMaterial` and `StarMeshComponent` are valid before using them.

### Expected result
Next code is just an example, but looks much more clear in my opinion:
```cpp
void APSStarActor::UpdateStarActorMeshMaterial(UMaterialInstanceDynamic* StarDynamicProgressMaterial, float AmountOfStars, bool bIsLockedStar)
{
    if (!ensureMsgf(StarDynamicProgressMaterial, TEXT("ASSERT: [%i] %hs:\n'StarDynamicProgressMaterial' is not valid!"), __LINE__, __FUNCTION__)
        || !ensureMsgf(StarMeshComponent, TEXT("ASSERT: [%i] %hs:\n'StarMeshComponent' is not valid!"), __LINE__, __FUNCTION__))
    {
        return;  // Early return if pointers are invalid
    }

    if (bIsLockedStar) 
    {
        StarMeshComponent->SetMaterial(0, UPSDataAsset::Get().GetLockedProgressionMaterial());
        return;  // Early return for locked stars
    }

    if (AmountOfStars > 0 && AmountOfStars < 1) 
    {
        StarMeshComponent->SetMaterial(0, StarDynamicProgressMaterial);
        StarDynamicProgressMaterial->SetScalarParameterValue(StarMaterialSlotName, AmountOfStars / StarMaterialFractionalDivisor);
        return;  // Early return for fractional stars
    }

    // Whole number stars
    StarMeshComponent->SetMaterial(0, UPSDataAsset::Get().GetUnlockedProgressionMaterial());
}
```

Reply:
agree, improved with: https://github.com/h4rdmol/ProgressionSystem/pull/17/commits/b7d4d4b3cb4a4016bd13bf127c8ebaf444b99b56